### ISSUE TITLE
wavesurfer not loading direct blob

### ISSUE THREAD

**Issue by interwap** [katspaugh/wavesurfer.js#904]
So I have a blob object which is a response from an ajax request. It's of type audio/mpeg. When I try to do wavesurfer.loadBlob it doesn't work although I can confirm I have a blob response. Any ideas?

**Comment by mspae** [comment]
Can you post a codepen or some code that reproduces this issues?

**Comment by interwap** [comment]
Sample code
`  
xmlhttp.onload = function(){
  
var audioFile = xmlhttp.response;

     var wavesurfer = WaveSurfer.create({
        container: '#waveform',
        waveColor: '#000',
        cursorColor: '#ffa200',
        cursorWidth: 2,
        backend: 'MediaElement',
        height: 20,
        pixelRatio: 1,
        hideScrollbar: true,
        progressColor: '#ffa200'
    });

    wavesurfer.loadBlob(audioFile);



}
xmlhttp.open('GET', '/psyrine/media/music/blah.mp3');
  xmlhttp.responseType = 'blob';
  xmlhttp.send(null);`

**Comment by mspae** [comment]
TL;DR You are using the `MediaElement` backend, use the default backend and just call load with the URL directly, no need to do the ajax yourself.

Wavesurfer.js provides two backends (ways in which the audio is provided to the browser):

1. `MediaElement` attaches an `<audio>` element (with the audio URL as source attribute) to the page. This allows us to play from a specific position without having to download the whole file beforehand (by using the browser's capabilities), this in turn allows us to pass in an array of peaks, which means we don't have to download the whole file to decode it and get the peaks (to display the waveform) → This is perfect for very large audio files.
2. `WebAudio` is the default backend, we provide a URL to wavesurfer.js, which is retrieved with AJAX (but we could also pass in a Blob or File object directly, if we want more control) and then decoded to get the peaks and play the audio file.

**Comment by mspae** [comment]
But this should definitely be documented here https://wavesurfer-js.org/docs/methods.html

**Comment by mspae** [comment]
But this should definitely be documented here https://wavesurfer-js.org/docs/methods.html

**Comment by mjunker** [comment]
I have the same issue. When I using loadBlob using backend MediaElement the loadDecodedBuffer method calls this.backend.load(buffer):

loadDecodedBuffer: function (buffer) {
        this.backend.load(buffer);

 The problem is that corresponding load function on in mediaelement.js expects the container as the second element and so the load function fails when executing this line (because container is null):

        var prevMedia = container.querySelector(this.mediaType);

But it looks like the load function of MediaElement was never meant to be used with blobs.

Is there a chance this will be supported with MediaElement or if this could be implemented? I can't use WebAudio as backend because I want to be able to slow down the audio without having pitch shift.

**Comment by katspaugh** [comment]
You can load a blob into a media-element via the [URL API](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL).

```
let audio = new Audio();
audio.src = URL.createObjectURL(blob);
wavesurfer.load(audio);
```

**Comment by mjunker** [comment]
This solved my problem. Thanks a lot for your help!

**Comment by alivtar** [comment]
In my case, i set the responseType of the request to be "blob" instead of converting the response to a Blob.
Then the response coming from API, was converted to a URL with `URL.createObjectURL(data_received_from_api)`

**Comment by ZeeshanAhmadKhalil** [comment]
In case someone is using `tauri.js` and trying to play `.mp3` from appDataDirectory (or some other directory) then this answer may be helpful.

https://stackoverflow.com/a/77191258/9108471
